### DISCUSSION TITLE
ffmpeg amix filter correpondant in libmedia

### DISCUSSION THREAD

**Discussion by Kuo-TingKai** [zhaohappy/libmedia#20] [Q&A]
Dear maintainer,
I would like to ask, is there any feature corresponding to ffmpeg-amix filter to fuse multiple audio source into one, something like this:
```bash
ffmpeg -i input1.mp3 -i input2.mp3 -i input3.mp3 -filter_complex amix=inputs=3:duration=longest output.mp3
```
I don't want to call the subprocess to run the above bash command in my ts file.
Thanks!

**Comment by zhaohappy** [comment]
libmedia does not yet implement the amix filter now. You can try to implement one yourself. The idea is to resample all PCM data to the same sample rate and then add them together as the output PCM data. I'll implement one later when I have time.
